src/test/java/SetupRoutesTest.java

```diff
@@ -1,22 +1,164 @@
 
 
 
+import http.handlers.HeadHandler;
+import http.helper.Handler;
+import http.helper.Utils;
+import http.request.Request;
+import http.response.Response;
 import http.router.Router;
 import http.router.SetupRouter;
 import org.junit.jupiter.api.Test;
 
-import java.util.Set;
+import java.io.ByteArrayOutputStream;
+import java.util.HashMap;
+import java.util.*;
 
 import static org.junit.jupiter.api.Assertions.*;
 
 public class SetupRoutesTest {
```

Comment:
👏  this is a pleasure to look at - a complete test suite for the setup with extracted helper functions to remove repetition. Awesome!

Reply:
Thank you Justyna 👍 😄